org.lflang.tests/src/org/lflang/tests/TestBase.java

```diff
@@ -572,6 +577,47 @@ static public String stackTraceToString(Throwable t) {
             exit 0
             """;
 
+    /**
+     * Returns true if arduino-cli exists, false otherwise.
+     */
+    private boolean checkArduinoCLIExists() {
+        LFCommand checkCommand = LFCommand.get("arduino-cli", List.of("version"));
+        return checkCommand != null && checkCommand.run() == 0;
+    }
+    
+    /**
+     * Return a ProcessBuilder used to test Arduino in a non-federated, unthreaded environment.
+     * See the following for references on arduino-cli:
+     * https://arduino.github.io/arduino-cli/
+     *
+     * @param test The test to get the execution command for.
+     */
+    private ProcessBuilder getArduinoExecCommand(LFTest test) {
```

Comment:
This does not belong in the test framework. Let's invoke `arduino-cli` as part of the build process (provided that the necessary target properties are provided). If they're missing, we should print a message telling the user how to obtain them. We can just set the test phase to `BUILD` as I had suggested, because, either way, the tests themselves will not be executed in CI.

Reply:
Done.